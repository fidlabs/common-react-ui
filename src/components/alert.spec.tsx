import { render } from "@testing-library/react";
import { InfoIcon } from "lucide-react";
import { describe, expect, it } from "vitest";
import { Alert, AlertDescription, AlertProps, AlertTitle } from "./alert";

type Variant = NonNullable<AlertProps["variant"]>;

const variants: Variant[] = ["default", "destructive", "error", "warning"];

describe("Button component", () => {
  it.each(variants)("renders correctly for variant %s", (variant) => {
    const { container } = render(
      <Alert variant={variant}>
        <InfoIcon />
        <AlertTitle>{variant} Alert</AlertTitle>
        <AlertDescription>
          Description of altert of variant {variant}
        </AlertDescription>
      </Alert>
    );
    expect(container).toMatchSnapshot();
  });
});
