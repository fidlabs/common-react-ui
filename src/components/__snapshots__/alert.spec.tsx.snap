// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button component > renders correctly for variant default 1`] = `
<div>
  <div
    class="relative w-full rounded-lg border p-4 [&>svg~*]:pl-7 [&>svg+div]:translate-y-[-3px] [&>svg]:absolute [&>svg]:left-4 [&>svg]:top-4 [&>svg]:text-foreground bg-background text-foreground"
    role="alert"
  >
    <svg
      class="lucide lucide-info"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="12"
        cy="12"
        r="10"
      />
      <path
        d="M12 16v-4"
      />
      <path
        d="M12 8h.01"
      />
    </svg>
    <h5
      class="mb-1 font-medium leading-none tracking-tight"
    >
      default
       Alert
    </h5>
    <div
      class="text-sm [&_p]:leading-relaxed"
    >
      Description of altert of variant 
      default
    </div>
  </div>
</div>
`;

exports[`Button component > renders correctly for variant destructive 1`] = `
<div>
  <div
    class="relative w-full rounded-lg border p-4 [&>svg~*]:pl-7 [&>svg+div]:translate-y-[-3px] [&>svg]:absolute [&>svg]:left-4 [&>svg]:top-4 border-destructive/50 text-destructive dark:border-destructive [&>svg]:text-destructive"
    role="alert"
  >
    <svg
      class="lucide lucide-info"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="12"
        cy="12"
        r="10"
      />
      <path
        d="M12 16v-4"
      />
      <path
        d="M12 8h.01"
      />
    </svg>
    <h5
      class="mb-1 font-medium leading-none tracking-tight"
    >
      destructive
       Alert
    </h5>
    <div
      class="text-sm [&_p]:leading-relaxed"
    >
      Description of altert of variant 
      destructive
    </div>
  </div>
</div>
`;

exports[`Button component > renders correctly for variant error 1`] = `
<div>
  <div
    class="relative w-full rounded-lg border p-4 [&>svg~*]:pl-7 [&>svg+div]:translate-y-[-3px] [&>svg]:absolute [&>svg]:left-4 [&>svg]:top-4 border-red-500 bg-red-100 [&>svg]:text-red-500"
    role="alert"
  >
    <svg
      class="lucide lucide-info"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="12"
        cy="12"
        r="10"
      />
      <path
        d="M12 16v-4"
      />
      <path
        d="M12 8h.01"
      />
    </svg>
    <h5
      class="mb-1 font-medium leading-none tracking-tight"
    >
      error
       Alert
    </h5>
    <div
      class="text-sm [&_p]:leading-relaxed"
    >
      Description of altert of variant 
      error
    </div>
  </div>
</div>
`;

exports[`Button component > renders correctly for variant warning 1`] = `
<div>
  <div
    class="relative w-full rounded-lg border p-4 [&>svg~*]:pl-7 [&>svg+div]:translate-y-[-3px] [&>svg]:absolute [&>svg]:left-4 [&>svg]:top-4 border-yellow-500 bg-yellow-100 [&>svg]:text-yellow-500"
    role="alert"
  >
    <svg
      class="lucide lucide-info"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="12"
        cy="12"
        r="10"
      />
      <path
        d="M12 16v-4"
      />
      <path
        d="M12 8h.01"
      />
    </svg>
    <h5
      class="mb-1 font-medium leading-none tracking-tight"
    >
      warning
       Alert
    </h5>
    <div
      class="text-sm [&_p]:leading-relaxed"
    >
      Description of altert of variant 
      warning
    </div>
  </div>
</div>
`;
